internal/config/configdomain/proposal_stack_lineage_builder.go

```diff
@@ -0,0 +1,131 @@
+package configdomain
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/git-town/git-town/v21/internal/forge/forgedomain"
+	"github.com/git-town/git-town/v21/internal/git/gitdomain"
+	. "github.com/git-town/git-town/v21/pkg/prelude"
+)
+
+func NewProposalStackLineageBuilder(args *ProposalStackLineageArgs) Option[ProposalStackLineageBuilder] {
+	if args == nil || args.Connector.IsNone() {
+		return None[ProposalStackLineageBuilder]()
+	}
+
+	if args.MainAndPerennialBranches.GetOrDefault().Contains(args.CurrentBranch) {
+		// cannot create proposal stack lineage for main or perennial branch
+		return None[ProposalStackLineageBuilder]()
+	}
+
+	findPropsalFn, hasFindPropsalFn := args.Connector.GetOrPanic().FindProposalFn().Get()
+	if !hasFindPropsalFn {
+		return None[ProposalStackLineageBuilder]()
+	}
+
+	lineage := args.Lineage.BranchLineage(args.CurrentBranch)
+	builder := &ProposalStackLineageBuilder{
+		mainAndPerennialBranches: args.MainAndPerennialBranches.GetOrDefault(),
+		orderedLineage:           make([]*proposalLineage, 0, len(args.Lineage.data)),
+	}
+	for _, currBranch := range lineage {
+		currBranchParent := args.Lineage.Parent(currBranch)
+		var err error
+		builder, err = builder.addBranch(currBranch, currBranchParent, findPropsalFn)
+		if err != nil {
+			fmt.Printf("failed to build proposal stack lineage: %s\n", err.Error())
+			return None[ProposalStackLineageBuilder]()
+		}
+	}
+
+	return Some(*builder)
+}
+
+type proposalLineage struct {
+	branch   gitdomain.LocalBranchName
+	proposal Option[forgedomain.ProposalData]
+}
+
+type ProposalStackLineageBuilder struct {
+	mainAndPerennialBranches gitdomain.LocalBranchNames
+	orderedLineage           []*proposalLineage
+}
+
+func (self *ProposalStackLineageBuilder) Build(args *ProposalStackLineageArgs) Option[string] {
```

Comment:
Return a non-optional string here. This function never returns the `None` variant.

```suggestion
func (self *ProposalStackLineageBuilder) Build(args *ProposalStackLineageArgs) string {
```